packages/flutter/lib/src/material/banner.dart

```diff
@@ -138,18 +179,101 @@ class MaterialBanner extends StatelessWidget {
   /// Defaults to [OverflowBarAlignment.end].
   final OverflowBarAlignment overflowAlignment;
 
+  /// The animation driving the entrance and exit of the material banner.
+  final Animation<double>? animation;
+
+  /// Called the first time that the material banner is visible within a [Scaffold].
+  final VoidCallback? onVisible;
```

Comment:
Same here - document what happens if this is null/default behavior. Is this still called if it is not used with the SacffoldMessenger?

Reply:
Nope, it's only called when the MaterialBanner is presented through the ScaffoldMessenger